test/dummy/app/models/widget_query.rb

```diff
@@ -5,6 +5,14 @@ class WidgetQuery
 
   self.percolates_model = Widget
 
+  doctype.mapping = {
+    properties: {
+      name: { type: 'keyword' },
+      color: { type: 'keyword' },
+      query: { type: 'percolator' }
+    }
+  }
+
```

Comment:
Without defining mapping, the Percolator Model in the test suite was being created with a very confusing, and seemingly incorrect, mapping. See image below:

<img width="1792" alt="screen shot 2018-03-13 at 1 23 13 pm" src="https://user-images.githubusercontent.com/7398833/37365214-78006c70-26c2-11e8-8268-b339f7481534.png">

I think explicit mapping may be required a majority of the time. Determining mapping dynamically could probably use some improvement.